lib/hijack/timeout_manager.js

```diff
@@ -0,0 +1,56 @@
+export const TimeoutManager = {
+  id: 0,
+  map: new Map(),
+
+  prettyMessage: {
+    method: (method) => `Method "${method}" still running after`,
+    sub: (sub) => `Subscription "${sub}" still running after`
+  },
+
+  addTimeout (fn, timeout) {
+    if (!fn) {
+      throw new Error('TimeoutManager.addTimeout: fn is required');
+    }
+
+    const id = ++this.id;
+
+    this.map.set(id, setTimeout(Meteor.bindEnvironment(() => {
+      fn();
+
+      this.map.delete(id);
+    }), timeout));
+
+    return id;
+  },
+
+  trackTimeout ({ kadiraInfo, msg, timeout = Kadira.options.stalledTimeout }) {
+    if (!timeout) {
+      return;
+    }
+
+    const type = msg.msg;
+    const method = msg.method || msg.name;
+
+    const error = new Error(`${this.prettyMessage[type](method) || 'Unknown Timeout'} ${timeout}ms`);
+
+    kadiraInfo.timeoutId = this.addTimeout(() => {
+      Kadira.EventBus.__emitter.emit('timeout', kadiraInfo, error);
```

Comment:
Does `Kadira.EventBus.emit` work?

Reply:
There is a wrapper around the emitter methods which change their signature

https://github.com/leonardoventurini/monti-apm-agent/blob/06cfe3b6848af5bed6ff89525c21ce6101201432/lib/common/unify.js#L31

I think can adjust it so it uses the same syntax